src/curvefit/solvers/solvers.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import scipy.optimize as sciopt
+
+from curvefit.core.effects2params import effects2params
+
+class ModelNotDefinedError(Exception):
+    pass
+
+class SolverNotDefinedError(Exception):
+    pass
+
+
+class Base:
+
+    def __init__(self, model_instance=None):
+        self.model = model_instance
+        self.x_opt = None
+        self.fun_val_opt = None
+
+    def set_model_instance(self, model_instance):
+        self.model = model_instance
+
+    def detach_model_instance(self):
+        self.model = None
+
+    def get_model_instance(self):
+        if self.model is not None:
+            return self.model
+        else:
+            raise ModelNotDefinedError()
+
+    def fit(self, data, x_init=None, options=None):
+        raise NotImplementedError()
+    
+    def predict(self, data):
```

Comment:
@jizezhang what is `data` here? Is is the data object or data specifications? And does it contain times that we want to predict for?

Reply:
so I was imagining it to be any data that would be needed for prediction. For `CoreModel` I think we just need `t`. But please let me know if you have any suggestions! I didn't think much about it and not sure if I like it either.

Reply:
I think it would be good to have `predict` stay similar to what it is right now -- so it needs a function argument (since it won't always be `self.model.forward`), a `group` and a `t` for time.

If we do that, it means that we also need a `predict` function for the model instance. I've commented above!

Reply:
hmm, so before I meant it to be a general function for solver class, not necessarily for `CoreModel` solvers. but now that I look at it, it does feel a bit weird and unnecessary here. maybe I'll just remove this function from base class, and define `predict` as what you commented above for `CoreModel` class?

Reply:
Yeah I think it's fine to leave `predict` out of the `Solver` class and keep it with the model instance